sql/core/src/test/resources/sql-tests/inputs/transform.sql

```diff
@@ -183,3 +189,126 @@ SELECT a, b, decode(c, 'UTF-8'), d, e, f, g, h, i, j, k, l FROM (
     NULL DEFINED AS 'NULL'
   FROM t
 ) tmp;
+
+SELECT TRANSFORM(b, a, CAST(c AS STRING))
+  USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM(1, 2, 3)
+  USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM(1, 2)
+  USING 'cat' AS (a INT, b INT)
+FROM script_trans
+LIMIT 1;
+
+SELECT TRANSFORM(
+  b AS d5, a,
+  CASE
+    WHEN c > 100 THEN 1
+    WHEN c < 100 THEN 2
+  ELSE 3 END)
+  USING 'cat' AS (a, b,  c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM(b, a, c + 1)
+  USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM(*)
+  USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
```

Comment:
These queries above are not supported in the current master?

Reply:
> These queries above are not supported in the current master?

Support, what's wrong? It have a correct answer.

Reply:
I just want to know why these tests are added in this PR... That's because these tests seems to be not related to aggregation/window/lateralView.

Reply:
> I just want to know why these tests are added in this PR... That's because these tests seems to be not related to aggregation/window/lateralView.

Yea, here want to show after this pr's change, each kind of expressions can work well, such as `alias`, `case when`, `binary compute` etc...